### PR TITLE
NetInfo for buyer app

### PR DESCRIPTION
**Pull request by onkarpandit**

### CODE REVIEW

File: android/src/main/java/com/reactlibrary/NetworkBroadcastReceiver.java [hunk 1]
```diff
@@ -25,6 +34,15 @@ public void onReceive(Context context, Intent intent) {
     }
 
     private void setConnectionStatus(boolean status) {
+        NetworkConnection network = new NetworkConnection();
```
Comment (Sunil18) [inline android/src/main/java/com/reactlibrary/NetworkBroadcastReceiver.java]: getInstance()

File: android/src/main/java/com/reactlibrary/NetworkBroadcastReceiver.java [hunk 2]
```diff
@@ -25,6 +34,15 @@ public void onReceive(Context context, Intent intent) {
...
+        boolean currentStatus = network.getConnectionStatus();
+        WritableMap receivedMessage = Arguments.createMap();
+        receivedMessage.putBoolean("statusChange", status);
+        if (mContext!=null && !status==currentStatus){
+            mContext
```
Comment (Sunil18) [inline android/src/main/java/com/reactlibrary/NetworkBroadcastReceiver.java]: Can avoid repeated getJSModule call.

**Review by Sunil18 — CHANGES_REQUESTED** [review]

File: android/src/main/java/com/reactlibrary/NetworkBroadcastReceiver.java [hunk 3]
```diff
@@ -25,6 +34,15 @@ public void onReceive(Context context, Intent intent) {
     }
 
     private void setConnectionStatus(boolean status) {
+        NetworkConnection network = new NetworkConnection();
+        boolean currentStatus = network.getConnectionStatus();
+        WritableMap receivedMessage = Arguments.createMap();
+        receivedMessage.putBoolean("statusChange", status);
```
Comment (manthan-25) [inline android/src/main/java/com/reactlibrary/NetworkBroadcastReceiver.java]: can change key to "status" or "isConnected"

File: android/src/main/java/com/reactlibrary/NetworkBroadcastReceiver.java [hunk 4]
```diff
@@ -2,17 +2,26 @@
...
 import android.net.ConnectivityManager;
 import android.net.NetworkInfo;
 
+import com.facebook.react.bridge.WritableMap;
+import com.facebook.react.modules.core.DeviceEventManagerModule;
+
 
 public class NetworkBroadcastReceiver extends BroadcastReceiver {
     private NetworkConnection netInfo = null;
+    private static final String EVENT_CHANGE = "net-handler:Status Change";
```
Comment (manthan-25) [inline android/src/main/java/com/reactlibrary/NetworkBroadcastReceiver.java]: can change to "connectionChange"

File: android/src/main/java/com/reactlibrary/NetworkBroadcastReceiver.java [hunk 5]
```diff
@@ -25,6 +34,15 @@ public void onReceive(Context context, Intent intent) {
     }
 
     private void setConnectionStatus(boolean status) {
+        NetworkConnection network = new NetworkConnection();
+        boolean currentStatus = network.getConnectionStatus();
+        WritableMap receivedMessage = Arguments.createMap();
+        receivedMessage.putBoolean("statusChange", status);
+        if (mContext!=null && !status==currentStatus){
```
Comment (manthan-25) [inline android/src/main/java/com/reactlibrary/NetworkBroadcastReceiver.java]: can use !=

File: android/src/main/java/com/reactlibrary/NetworkBroadcastReceiver.java [hunk 6]
```diff
@@ -2,17 +2,30 @@
...
 
+import com.facebook.react.bridge.WritableMap;
+import com.facebook.react.modules.core.DeviceEventManagerModule;
+
 
 public class NetworkBroadcastReceiver extends BroadcastReceiver {
     private NetworkConnection netInfo = null;
+    private static final String EVENT_CHANGE = "connectionChange";
+    private DeviceEventManagerModule.RCTDeviceEventEmitter jsModuleEventEmitter = null;
+    private mContext = null;
```
Comment (manthan-25) [inline android/src/main/java/com/reactlibrary/NetworkBroadcastReceiver.java]: should be private ReactApplicationContext mcontext

File: android/src/main/java/com/reactlibrary/NetworkBroadcastReceiver.java [hunk 7]
```diff
@@ -25,6 +36,18 @@ public void onReceive(Context context, Intent intent) {
     }
 
     private void setConnectionStatus(boolean status) {
+        boolean currentStatus = netInfo.getConnectionStatus();
```
Comment (Sunil18) [inline android/src/main/java/com/reactlibrary/NetworkBroadcastReceiver.java:37]: null check for netInfo

**Review by Sunil18 — CHANGES_REQUESTED** [review]

**Review by Sunil18 — APPROVED** [review]

**Pull request merged**